packages/datetime/src/dateInput.tsx

```diff
@@ -447,10 +464,37 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
             // the page. tabbing forward should *not* close the popover, because
             // focus will be moving into the popover itself.
             this.setState({ isOpen: false });
+        } else if (e.which === Keys.ESCAPE) {
```

Comment:
popovers have a feature to `closeOnEscapeKey`... can we leverage that instead?

try adding an `onInteraction` handler to the `Popover`.

Reply:
- [ ] use `closeOnEscapeKey` on the popover

Reply:
Looks like without additional work, pressing ESC while **focus is within the popover**, the popover will close.

However, when the **focus is on the input** we still need to handle ESC and set the popover to close.